packages/cli/src/cmds/assert/assert.ts

```diff
@@ -0,0 +1,105 @@
+import process from 'process';
+import { verbose } from '../../utils';
+import ValidationError from '../error/validationError';
+import AbortError from '../error/abortError';
+import * as fs from 'fs';
+import 'fs/promises';
+import * as console from 'console';
+import { buildAppMap } from '../../search/utils';
+import AssertionChecker from './assertionChecker';
+import Assertion from './assertion';
+import { glob } from 'glob';
+import AssertionsConfig from './assertionsConfig';
+import ProgressFormatter from './formatter/progressFormatter';
+import PrettyFormatter from './formatter/prettyFormatter';
+
+exports.command = 'assert';
+exports.describe = 'Run assertions for AppMaps in the directory';
+
+exports.builder = (args) => {
+  args.option('appmap-dir', {
+    describe: 'directory to recursively inspect for AppMaps',
+    default: 'tmp/appmap',
+  });
+  args.option('config', {
+    describe: 'path to assertions config file',
+    default: './assertionsConfig',
+    alias: 'c',
+  });
+  args.option('format', {
+    describe: 'output format (progress, pretty)',
+    default: 'progress',
+    alias: 'f',
+  });
+  return args.strict();
+};
+
+exports.handler = async (argv) => {
+  verbose(argv.verbose);
+
+  const commandFn = async () => {
+    const { appmapDir, config, format } = argv;
+
+    if (!fs.existsSync(appmapDir)) {
```

Comment:
Prefer asynchronous

Reply:
If there's anywhere it's resonable to do it, it's in setup code like this though :-)